app/Http/Controllers/V2/Dashboard/ViewTreeRestorationGoalController.php

```diff
@@ -0,0 +1,187 @@
+        $siteIds = $this->getSiteIds($allProjectIds);
+        $distinctDates = $this->getDistinctDates($siteIds);
+        $latestDueDate = $this->getLatestDueDate($distinctDates);
+
+        $forProfitProjectIds = $this->filterProjectIdsByType($rawProjectIds, 'for-profit-organization');
+        $nonProfitProjectIds = $this->filterProjectIdsByType($rawProjectIds, 'non-profit-organization');
+        $forProfitSiteIds = $this->getSiteIds($forProfitProjectIds);
+        $nonProfitSiteIds = $this->getSiteIds($nonProfitProjectIds);
+
+        $forProfitTreeCount = $this->treeCountByDueDate($forProfitProjectIds, $latestDueDate["year"], $latestDueDate["month"]);
+        $nonProfitTreeCount = $this->treeCountByDueDate($nonProfitProjectIds, $latestDueDate["year"], $latestDueDate["month"]);
+
+        $totalTreesGrownGoal = Project::sum('trees_grown_goal');
+
+        $treesUnderRestorationActualTotal = $this->treeCountPerPeriod($siteIds, $distinctDates);
+        $treesUnderRestorationActualForProfit = $this->treeCountPerPeriod($forProfitSiteIds, $distinctDates);
+        $treesUnderRestorationActualNonProfit = $this->treeCountPerPeriod($nonProfitSiteIds, $distinctDates);
+
+        $averageSurvivalRateTotal = $this->getAverageSurvival($allProjectIds);
+        $averageSurvivalRateForProfit = $this->getAverageSurvival($forProfitProjectIds);
+        $averageSurvivalRateNonProfit = $this->getAverageSurvival($nonProfitProjectIds);
+
+        $result = [
+            'forProfitTreeCount' => (int) $forProfitTreeCount,
+            'nonProfitTreeCount' => (int) $nonProfitTreeCount,
+            'totalTreesGrownGoal' => (int) $totalTreesGrownGoal,
+            'treesUnderRestorationActualTotal' => $treesUnderRestorationActualTotal,
+            'treesUnderRestorationActualForProfit' => $treesUnderRestorationActualForProfit,
+            'treesUnderRestorationActualNonProfit' => $treesUnderRestorationActualNonProfit,
+            'averageSurvivalRateTotal' => floatval($averageSurvivalRateTotal),
+            'averageSurvivalRateForProfit' => floatval($averageSurvivalRateForProfit),
+            'averageSurvivalRateNonProfit' => floatval($averageSurvivalRateNonProfit),
+        ];
+
+        return new JsonResponse($result);
+    }
+
+    private function prepareProjectQuery(Request $request, array $organizationTypes)
+    {
+        $query = Project::join('organisations', 'v2_projects.organisation_id', '=', 'organisations.id')
+            ->where('v2_projects.framework_key', '=', 'terrafund')
+            ->whereIn('organisations.type', $organizationTypes);
+
+        if ($request->has('country')) {
+            $country = $request->input('country');
+            $query->where('country', $country);
+        } elseif ($request->has('uuid')) {
+            $projectUuid = $request->input('uuid');
+            $query->where('v2_projects.uuid', $projectUuid);
+        }
+
+        return $query;
+    }
```

Comment:
I've noted that this is a common pattern from a previous PR, can you reuse the code with a helper, even try to review `TerrafundDashboardQueryHelper` and see if you can reuse that function or add this function there to be used in multiple places.